hub/constraints/instruction-handling/acc.lisp

```diff
@@ -47,135 +71,171 @@
+                                           (if-zero (account-instruction---is-CODESIZE)
                                                     ;; DEC_FLAG_4 = 0
                                                     (begin
-                                                      (eq!  (shift account/ADDRESS_HI  2) (account-instruction-account-address-hi))
-                                                      (eq!  (shift account/ADDRESS_LO  2) (account-instruction-account-address-lo)))
+                                                      (eq!  (shift account/ADDRESS_HI    ROFF_ACC___ACCOUNT_READING_ROW) (account-instruction---account-address-hi))
+                                                      (eq!  (shift account/ADDRESS_LO    ROFF_ACC___ACCOUNT_READING_ROW) (account-instruction---account-address-lo)))
                                                     ;; DEC_FLAG_4 = 1
                                                     (begin
-                                                      (eq!  (shift account/ADDRESS_HI  2)  (account-instruction-byte-code-address-hi))
-                                                      (eq!  (shift account/ADDRESS_LO  2)  (account-instruction-byte-code-address-lo)))))))))
-
-(defconstraint   account-instruction-value-constraints                                      (:guard (account-instruction-unexceptional))
-                 (begin
-                   (if-not-zero [ stack/DEC_FLAG 1 ]
-                                (begin
-                                  (eq!  [ stack/STACK_ITEM_VALUE_HI 4 ] 0)
-                                  (eq!  [ stack/STACK_ITEM_VALUE_LO 4 ] (shift account/BALANCE 1))))
-                   ;;
-                   (if-not-zero [ stack/DEC_FLAG 2 ]
-                                (begin
-                                  (eq!  [ stack/STACK_ITEM_VALUE_HI 4 ] 0)
-                                  (eq!  [ stack/STACK_ITEM_VALUE_LO 4 ] (shift  (* account/CODE_SIZE account/HAS_CODE) 1))))
-                   ;;
-                   (if-not-zero [ stack/DEC_FLAG 3 ]
-                                (begin
-                                  (eq!  [ stack/STACK_ITEM_VALUE_HI 4 ] (shift  (* account/CODE_HASH_HI account/EXISTS)  1))
-                                  (eq!  [ stack/STACK_ITEM_VALUE_LO 4 ] (shift  (* account/CODE_HASH_LO account/EXISTS)  1))))
-                   ;;
-                   (if-not-zero [ stack/DEC_FLAG 4 ]
-                                (begin
-                                  (eq!  [ stack/STACK_ITEM_VALUE_HI 4 ] 0)
-                                  (eq!  [ stack/STACK_ITEM_VALUE_LO 4 ] (shift account/CODE_SIZE 2))))
-                   ;;
-                   (if-not-zero (- 1 (account-instruction-decoded-flags-sum))
-                                (begin
-                                  (eq!  [ stack/STACK_ITEM_VALUE_HI 4 ] 0)
-                                  (eq!  [ stack/STACK_ITEM_VALUE_LO 4 ] (shift account/CODE_SIZE 2))))))
```

Comment:
**Bug fix.** We were using the `code size` rather than the `balance` as the return value of `SELFBALANCE`.